notice.go

```diff
@@ -63,17 +66,84 @@ func (n *Notice) String() string {
 	return fmt.Sprintf("Notice<%s: %s>", e.Type, e.Message)
 }
 
+// stackTraces returns the stackTrace of an error.
+// It is part of the errors package public interface.
+type stackTracer interface {
+	StackTrace() errors.StackTrace
+}
+
+// getStack returns the stacktrace associated with e. If e is an
+// error from the errors package its stacktrace is extracted, otherwise
+// the current stacktrace is collected end returned.
+func getStack(e interface{}, depth int) []StackFrame {
+	if err, ok := e.(stackTracer); ok {
+		return stackFromErrorWithStackTrace(err)
+	}
+
+	return stack(depth)
+}
+
+// parseFrame parses and errors.Frame and returns a non-nil StackFrame
+// if the parsing is successful.
+//
+// We need to parse a formatted output for the func. name/file name/line no.,
+// because Frame does not have public methods to access these values, only
+// the output is specified.
+func parseFrame(f *errors.Frame) *StackFrame {
+	// A stack frame can be formatted several ways:
+	//    %+s    func\nfile
+	//    %d    source line
+	buf := fmt.Sprintf("%+s\n%d", f, f)
+	parts := strings.Split(buf, "\n")
+	if len(parts) != 3 {
+		return nil
+	}
+
+	fn := strings.TrimSpace(parts[0])
+	file := strings.TrimSpace(parts[1])
+	line, err := strconv.ParseInt(parts[2], 10, 64)
+	if err != nil {
+		return nil
+	}
+	return &StackFrame{file, int(line), fn}
+}
+
+// stackFromErrorWithStackTrace extracts the stacktrace from e.
+func stackFromErrorWithStackTrace(e stackTracer) []StackFrame {
+	var frames []StackFrame
+	for _, f := range e.StackTrace() {
+		pf := parseFrame(&f)
+		if pf != nil {
+			frames = append(frames, *pf)
+		}
+	}
+	// Remove the frames of runtime.main and runtime.goexit
+	frames = frames[:len(frames)-2]
```

Comment:
Let's omit this at least for now? First we didn't remove these frames until now and second Go for some reason decided to show them...